internal/controller/controllers/agent_controller.go

```diff
@@ -1279,6 +1303,12 @@ func (r *AgentReconciler) updateIfNeeded(ctx context.Context, log logrus.FieldLo
 		InfraEnvID:       internalHost.InfraEnvID,
 		HostUpdateParams: &models.HostUpdateParams{},
 	}
+
+	hostUpdate, err = r.updateNodeLabels(log, internalHost, agent, params)
```

Comment:
suggesting to use a different parameter in case someone will add more code before this function he may forget that you are reassigning the value.

Reply:
OK